File: November 2024 Leetcode Solution/November-22.java
class Solution {
    public int maxEqualRowsAfterFlips(int[][] matrix) {
        Map<String, Integer> patternCount = new HashMap<>();
        int maxCount = 0;

        for (int[] row : matrix) {
            StringBuilder pattern = new StringBuilder();
            int firstElement = row[0];

            for (int val : row) {
                pattern.append((val ^ firstElement) == 0 ? '0' : '1');
            }

            patternCount.put(pattern.toString(), patternCount.getOrDefault(pattern.toString(), 0) + 1);
            maxCount = Math.max(maxCount, patternCount.get(pattern.toString()));
        }

        return maxCount;
    }
}

File: November 2024 Leetcode Solution/November-22.py
from collections import defaultdict
from typing import List

class Solution:
    def maxEqualRowsAfterFlips(self, matrix: List[List[int]]) -> int:
        count = defaultdict(int)

        for row in matrix:
            row_key = tuple(row) if row[0] == 0 else tuple(1 - n for n in row)
            count[row_key] += 1

        return max(count.values()) 
